src/server/engine_shard.cc

```diff
@@ -660,13 +660,24 @@ void EngineShard::Heartbeat() {
 
   CacheStats();
 
+  // TODO: iterate over all namespaces
+  DbSlice& db_slice = namespaces->GetDefaultNamespace().GetDbSlice(shard_id());
+  // Skip heartbeat if we are serializing a big value
+  static auto start = std::chrono::system_clock::now();
+  if (db_slice.WillBlockOnJournalWrite()) {
+    const auto elapsed = std::chrono::system_clock::now() - start;
+    if (elapsed > std::chrono::seconds(1)) {
+      LOG(WARNING) << "Stalled heartbeat() fiber for " << elapsed.count()
```

Comment:
you should use log ever t
as if the heartbeat is stalled due to big value writings you we will be flooded with this prints